packages/core/tests/error.test.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  DiagnosticError,
+  isDiagnosticError,
+  Loggable,
+  Log,
+  LogOptions,
+  LogLevel,
+} from '..';
+
+const content: Loggable = ['This is what happened', 'and this'];
+
+describe('DiagnosticError', () => {
+  const title = 'An error happened';
+  const suggestion: Loggable = 'This is what you can do about it';
+  it('defines a DiagnosticError', () => {
```

Comment:
I feel like this test is doing too much and can be broken down in the a number of different tests. 

* One test for `isDiagnosticError` behaviour with all it's respective assertions
* One test for verifying it's association with the `Error` class with it's respective assertions

I don't think there's much value in having the assertions for `title`, `content`, `suggestion` since the class doesn't do anything with it. 

`expect(err.message).toMatch(title);` seems ok. Lets readers know that  the `title` property gets propagated to the Error constructor.